### PR TITLE
Rename to citadel

### LINKED ISSUE #22: Change project name to lazaret
And update all the paths and such.

### PR DESCRIPTION
**Pull request by enj**
Fixes #22

@npmccallum I will rename the repo after this merges.

cc @lancejjohnson

### CODE REVIEW

File: pkg/kms/kms.go [hunk 1]
```diff
@@ -3,22 +3,19 @@ package kms
...
 )
 
-const (
-	version        = "v1beta1"
-	runtimeName    = "kms_cmd"
-	runtimeVersion = "0.0.1" // TODO embed git sha on build
-)
+const kmsAPIVersion = "v1beta1"
 
 var apiVersionResponse = &v1beta1.VersionResponse{
```
Comment (benjaminapetersen) [inline pkg/kms/kms.go:15]: I assume this is what you are renaming later?

Comment (benjaminapetersen) [reply pkg/kms/kms.go:15]: as its still `kms` in this file.

Comment (enj) [reply pkg/kms/kms.go:15]: I actually plan to make a `v1beta1` folder and, in the future, a `v1` folder (so a single daemon process will support all versions of the KMS API).

Comment (enj) [reply pkg/kms/kms.go:15]: This `kms` is correct because it refers to the Kube API only.

Comment (benjaminapetersen) [reply pkg/kms/kms.go:15]: +1 for the rename, then.

😄

File: pkg/cmd/citadel/options.go [hunk 2]
```diff
@@ -48,7 +48,7 @@ var (
 )
 
 func init() {
-	flag.StringVar(&args.endpoint, "endpoint", "", `the listen address (ex. unix:///tmp/kms.sock)`)
+	flag.StringVar(&args.endpoint, "endpoint", "", `the listen address (ex. unix:///tmp/socketfile.sock)`)
```
Comment (npmccallum) [inline pkg/cmd/citadel/options.go]: This undoes my previous change. I'm trying to keep doc strings short (for an 80 character terminal). Also, "socketfile.sock" is redundant.

File: README.md [hunk 3]
```diff
@@ -1,21 +1,21 @@
-# Welcome to KMS!
+# Welcome to Citadel!
 
-KMS is a simple daemon that implements the Kubernetes Key Management Service
+Citadel (c5l) is a simple daemon that implements the Kubernetes Key Management Service (KMS)
```
Comment (npmccallum) [inline README.md:3]: I don't like the `c5l` abbreviation. `citadel` isn't that long.

File: README.md [hunk 4]
```diff
@@ -29,7 +29,7 @@ socket activation is assumed.
 
 ### Optional
 
- * `--endpoint string`: the listen address (ex. `unix:///tmp/kms.sock`)
+ * `--endpoint string`: the listen address (ex. `unix:///tmp/socketfile.sock`)
```
Comment (npmccallum) [inline README.md]: See my previous comment about length and redundancy.

File: pkg/cmd/citadel/options.go (hunk 2 above)
Comment (enj) [reply pkg/cmd/citadel/options.go]: Trying to remove the use of "kms", and this was what the kube examples used.  I can pick something else that is short.

File: README.md (hunk 3 above)
Comment (enj) [reply README.md:3]: I used it as the short name in the prefix code (to mimic kube), so I wanted to actually use it so people would know what it was referring to.

I am not really worried about the length in the docs, it is more of "google will find it"

Comment (npmccallum) [reply README.md:3]: I still don't like it, but if you want to mimic kube, I can accept that.

**Comment by enj** [comment]
@npmccallum PTAL

**Review by npmccallum — APPROVED** [review]

**Pull request merged**